### PR TITLE
[CS-4019] Tests to unlock screen hook

### PR DESCRIPTION
**Pull request by oidoug**
### Description

Add a comprehensive list of tests to Unlock Screen.

I had difficulties getting the `getPin` return to properly reflect in the hook before checking values. Eventually, I figure it out, but it took a while 😓 

- [x] Completes #(CS-4019)

### CODE REVIEW

File: cardstack/src/screens/UnlockScreen/__tests__/useUnlockScreen.test.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    mockBiometryAvailableHelper();
+    mockAuthAuthorizedHelper();
+    mockGetPinHelper();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should authorized be called when inputed PIN match stored one', async () => {
```
Comment (danibonilha) [inline cardstack/src/screens/UnlockScreen/__tests__/useUnlockScreen.test.ts]: nit: the it, it's a helper to make the test case readability better, so we can reorder the words here, `it('should call setAuthorized when inputed PIN matches stored one')`

File: cardstack/src/screens/UnlockScreen/__tests__/useUnlockScreen.test.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    act(() => {
+      result.current.setInputPin(PIN);
+    });
+
+    expect(result.current.inputPin).toMatch(PIN);
+    expect(result.current.pinInvalid).toBeFalsy();
+    expect(mockSetAuthorized).toBeCalledTimes(1);
+  });
+
+  it('should PIN be invalid and authorized not be called when inputed PIN doest match stored one', async () => {
```
Comment (danibonilha) [inline cardstack/src/screens/UnlockScreen/__tests__/useUnlockScreen.test.ts]: `it('should not call setAuthorized if inputed PIN does not match stored one and set PIN as invalid')`

File: cardstack/src/screens/UnlockScreen/__tests__/useUnlockScreen.test.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    await waitFor(() => expect(getPin).toHaveBeenCalled());
+
+    act(() => {
+      result.current.setInputPin(wrongPIN);
+    });
+
+    expect(result.current.inputPin).toMatch('');
+    expect(result.current.pinInvalid).toBeTruthy();
+    expect(mockSetAuthorized).toBeCalledTimes(0);
```
Comment (danibonilha) [inline cardstack/src/screens/UnlockScreen/__tests__/useUnlockScreen.test.ts]: FYI: we can also use `expect(mockSetAuthorized).not.toBeCalled();`

File: cardstack/src/screens/UnlockScreen/useUnlockScreen.tsx [hunk 4]
```diff
@@ -97,11 +99,6 @@ export const useUnlockScreen = () => {
...
   }, [authenticateBiometrically, isBiometryEnabled]);
 
-  const retryBiometricLabel = useMemo(
-    () => `Try ${biometryLabel || 'device authentication'} again`,
-    [biometryLabel]
-  );
-
   return {
     inputPin,
     setInputPin,
```
Comment (danibonilha) [inline cardstack/src/screens/UnlockScreen/useUnlockScreen.tsx:107]: I believe we should drop the changes on this file, besides the reordering stuff

Comment (oidoug) [reply cardstack/src/screens/UnlockScreen/useUnlockScreen.tsx:107]: Thanks for the pointer, I forgot to revert those 👍

**Review by danibonilha — APPROVED** [review]
Way to go!

**Pull request merged**